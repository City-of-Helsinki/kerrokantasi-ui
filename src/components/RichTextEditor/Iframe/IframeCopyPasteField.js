import React from 'react';
import PropTypes from 'prop-types';
import getMessage from '../../../utils/getMessage';
import { parseIframeHtml, convertStyleDimensionSettings } from './IframeUtils';


class IframeCopyPasteField extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      htmlCopyPaste: '',
    };

    this.handleCopyPasteChange = this.handleCopyPasteChange.bind(this);
  }

  handleCopyPasteChange(event) {
    const {value} = event.target;
    this.setState({htmlCopyPaste: value});

    // if iframe has width and/or height in the style attribute
    // convert the dimensions into their own attributes
    const attributes = convertStyleDimensionSettings(parseIframeHtml(value));
    this.props.updateAttributes(attributes);
  }

  render() {
    return (
      <div className="input-container html-copy-paste-input">
        <label htmlFor="iframe-html-copy-paste">{getMessage("iframeHtmlCopyPaste")}</label>
        <textarea
            id="iframe-html-copy-paste"
            name="htmlCopyPaste"
            className="form-control"
            onChange={this.handleCopyPasteChange}
            value={this.state.htmlCopyPaste}
        />
      </div>
    );
  }
}

IframeCopyPasteField.propTypes = {
  updateAttributes: PropTypes.func,
};

export default IframeCopyPasteField;
