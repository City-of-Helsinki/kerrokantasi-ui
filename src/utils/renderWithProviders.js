/* eslint react/prop-types: 0 */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';
import { IntlProvider } from 'react-intl';

import createStore from "../createStore";
import { userOidcConfig, apiTokenClientConfig } from './oidcConfig';

const loginProviderConfig = {
  userManagerSettings: userOidcConfig,
  apiTokensClientSettings: apiTokenClientConfig,
  sessionPollerSettings: { pollIntervalInMs: 10000 },
}

const renderWithProviders = (
  ui,
  { preloadedState = {}, locale = 'fi', history, store = createStore(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider {...loginProviderConfig}>
      <IntlProvider locale={locale}>
        <Provider store={store}>{children}</Provider>
      </IntlProvider>
    </LoginProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};


export default renderWithProviders;
