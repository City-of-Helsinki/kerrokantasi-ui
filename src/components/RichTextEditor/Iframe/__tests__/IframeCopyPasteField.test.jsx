import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import renderWithProviders from '../../../../utils/renderWithProviders';
import IframeCopyPasteField from '../IframeCopyPasteField';

const renderComponent = (propOverrides) => {
  const props = {
    updateAttributes: vi.fn(),
    ...propOverrides,
  };

  return renderWithProviders(
    <MemoryRouter>
      <IframeCopyPasteField {...props} />
    </MemoryRouter>,
  );
};

describe('<IframeSelectField />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('should handleCopyPasteChange', async () => {
    renderComponent();

    const input = await screen.findByRole('textbox');

    const user = userEvent.setup();

    user.type(input, 'test');

    await waitFor(() => expect(input.value).toBe('test'));
  });
});
