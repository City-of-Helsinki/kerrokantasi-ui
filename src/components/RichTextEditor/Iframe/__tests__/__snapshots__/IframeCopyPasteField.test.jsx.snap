// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<IframeSelectField /> > should render correctly 1`] = `
<div>
  <div
    class="input-container html-copy-paste-input"
  >
    <label
      for="iframe-html-copy-paste"
    >
      Iframe leikkaa ja liit√§ html
    </label>
    <textarea
      class="form-control"
      id="iframe-html-copy-paste"
      name="htmlCopyPaste"
    />
  </div>
</div>
`;
